docs/tokens.md

```diff
@@ -3,137 +3,239 @@ id: tokens
+        _setTokenURI(newItemId, tokenURI);
+
+        return newItemId;
+    }
 }
 ```
 
-`ERC20Mintable` allows to add minters via `addMinter(addr)`, so they (like the DOGGO Network multisig) can mint tokens to the dogsitters in exchange for watching the nice doggos while their owners leave for vacation. The token is `ERC20Burnable` we want to have the ability to stake DOGGO tokens on our reputation—if the dogsitter does a bad job, their tokens get burned!
+The [`ERC721Full`](api/token/ERC721#erc721full) contract includes all standard extensions, and is probably the one you want to use. In particular, it includes [`ERC721Metadata`](api/token/ERC721#erc721metadata), which provides the [`_setTokenURI`](api/token/ERC721#ERC721Metadata._setTokenURI(uint256,string)) method we use to store an item's metadata.
 
-### A Note on `decimals`
+Also note that, unlike ERC20, ERC721 lacks a `decimals` field, since each token is distinct and cannot be partitioned.
 
-You might remember from the previous chapter about crowdsales about how math is performed in financial situations: **all currency math is done in the smallest unit of that currency**.
+New items can not be created:
 
-That means that the `totalSupply` of a token is actually in what we call `TKNbits`, not what you see as `TKN`. So if my total supply is `1` and we have `5` decimals in the token, that's actually `1 TKNbit` and will be displayed as `0.00001 TKN`.
+```javascript
+> GameItem.awardItem(playerAddress, "https://game.com/item-id-8u5h2m.json")
+7
+```
 
-You probably want to use a decimals of `18`, just like Ether, unless you have a special reason not to, so when you're minting tokens to people or transferring them around, you're actually sending the number `numTKN * 10^(decimals)`. So if I'm sending you `5` tokens using a token contract with 18 decimals, the method I'm executing actually looks like `transfer(yourAddress, 5 * 10^18)`.
+And the owner and metadata of each item queried:
+```javascript
+> GameItem.ownerOf(7)
+playerAddress
+> GameItem.tokenURI(7)
+"https://game.com/item-id-8u5h2m.json"
+```
 
-## ERC721
+This `tokenURI` should resolve to a JSON document that might look something like:
+
+```json
+{
+    "name": "Thor's hammer",
+    "description": "Mjölnir, the legendary hammer of the Norse god of thunder.",
+    "image": "https://game.com/item-id-8u5h2m.png",
+    "strength": 20
+}
+```
+
+For more information about the `tokenURI` metadata JSON Schema, check out the [ERC721 specification](https://eips.ethereum.org/EIPS/eip-721).
+
+_Note: you'll notice that the item's information is included in the metadata, but that information isn't on-chain! So a game developer could change the underlying metadata, changing the rules of the game! If you'd like to put all item information on-chain, you can extend ERC721 to do so (though it will be rather costly). You could also leverage IPFS to pin the tokenURI information, but these techniques are out of the scope of this overview guide._
```

Comment:
Is "pinning" the right term here? It has a specific meaning in IPFS.

An alternative term would be "commit", as in "commit and reveal".

Reply:
Tbh I have pretty much zero knowledge on how IPFS works, if you think 'commit' works better, let's go with that one.

Reply:
I would use the word “store”. Pin means to hold on to IPFS content.